### PR TITLE
Create parameter type aliases for scalar field types.

### PR DESCRIPTION
**Pull request by tseaver**
See: #3364

### CODE REVIEW

**Comment by dhermes** [comment]
I'll leave this review to @jonparrott.

Is there any reason to add docstrings?

File: spanner/google/cloud/spanner/__init__.py [hunk 1]
```diff
@@ -18,6 +18,13 @@
 import pkg_resources
 __version__ = pkg_resources.get_distribution('google-cloud-spanner').version
 
+from google.cloud.spanner._helpers import BOOL_PARAM_TYPE
```
Comment (jonparrott) [inline spanner/google/cloud/spanner/__init__.py]: I feel weird about plucking members out of a private module to expose. Do we have a `spanner.types` module?

Comment (tseaver) [inline spanner/google/cloud/spanner/__init__.py]: Seems like a perfectly reasonable use of https://en.wikipedia.org/wiki/Facade_pattern to me:  the "implementation" is private, but the things themselves are exposed to the public.

Comment (tseaver) [inline spanner/google/cloud/spanner/__init__.py]: And no, there is no extant `google.cloud.spanner.types` module.

Comment (jonparrott) [inline spanner/google/cloud/spanner/__init__.py]: @tseaver do you think it's worthwhile for us to introduce a `types` module to hold these constants? (like we've done with gapic clients like vision, speech, etc.)

Comment (lukesneeringer) [inline spanner/google/cloud/spanner/__init__.py]: I would like to move in the direction of adding a `types` module for everything; I had not done it for Spanner yet.

@tseaver Any objections to my doing that if I can do it today or tomorrow?

Comment (tseaver) [inline spanner/google/cloud/spanner/__init__.py]: fa34a34 adds the module here.  I'm not sure what else you have in mind for it.

File: spanner/google/cloud/spanner/types.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Types exported from this package."""
+
+from google.cloud.proto.spanner.v1 import type_pb2
+
+
+# Scalar paramter types
+STRING_PARAM_TYPE = type_pb2.Type(code=type_pb2.STRING)
```
Comment (lukesneeringer) [inline spanner/google/cloud/spanner/types.py]: Ideally, I would like to use the identical pattern we are using elsewhere, which has a few nice usability wins:

  1. It uses the identical type names to the `_pb2` modules.
  2. It rewrites `__module__` which helps users when looking at docs.
  3. It grabs all types from the modules on the fly, so if Spanner introduces new types, it does not need manual updating.

[Here](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/vision/google/cloud/vision_v1/types.py#L23-L35) is what I did in Vision. Would you have any objection to doing that here?

Comment (lukesneeringer) [inline spanner/google/cloud/spanner/types.py]: Derp, I just realized that you are _instantiating_ `type_pb2.Type` with constants.

**Comment by jonparrott** [comment]
Note that we don't expose protos directly in spanner. These items are a
special case where we can't map Python types to Spanner types so we need
some constant and using the pb2 Types is very convenient.

**Comment by lukesneeringer** [comment]
Yep, I realized my mistake.

**Comment by tseaver** [comment]
@lukesneeringer Any reason not to merge?

**Comment by lukesneeringer** [comment]
Merge away.

**Pull request merged**